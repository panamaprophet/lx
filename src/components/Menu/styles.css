.root {
    display: flex; 
    padding: 0;
    list-style: none;

    & .root {
        flex-flow: column;
    }
}

.position_type_fixed {
    position: fixed;
}

.position_type_absolute {
    position: absolute;
}

.position_type_relative {
    position: relative;
}

.position_type_static {
    position: static;
}