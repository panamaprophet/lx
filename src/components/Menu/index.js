import React from 'react';
import cx from 'classnames';

import MenuItem from '../MenuItem';

import styles from './styles.css';


const Menu = ({
    items,
    activeItemId,
    onClick = () => {},
    classNames = [],
    position = 'fixed'
}) => (
    <ul className={cx([
        styles.root,
        styles[`position_type_${position}`],
        ...classNames
    ])}>
        {items && items.map(item => (
            <MenuItem
                {...item}
                isActive={activeItemId === item.id}
                SubMenuComponent={Menu}
                onClick={onClick}
                key={item.id}
            />))}
    </ul>
);

export default Menu;
