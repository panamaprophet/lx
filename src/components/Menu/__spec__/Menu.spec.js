import React from 'react';
import {shallow} from 'enzyme';

import Menu from '../index';
import itemsMock from './mock';

describe('Menu', () => {
    test('component renders with default props', () => {
        const component = shallow(<Menu />);

        expect(component).toMatchSnapshot();
    });

    test('component renders with fixed position by default', () => {
        const component = shallow(<Menu items={itemsMock} />);
        const hasClass = component.hasClass('position_type_fixed');

        expect(hasClass).toBe(true);
    });

    test('component renders with specified position', () => {
        const component = shallow(<Menu 
            items={itemsMock}
            position="absolute"
        />);

        const hasClass = component.hasClass('position_type_absolute');

        expect(hasClass).toBe(true);
    });

    test('all passed classNames applied to component', () => {
        const expectedClassNames = ['additional', 'classnames'];

        const component = shallow(<Menu 
            classNames={expectedClassNames}
        />);

        const hasClass = expectedClassNames
            .reduce((result, className) => 
                result && component.hasClass(className), true);

        expect(hasClass).toBe(true);
    })
});