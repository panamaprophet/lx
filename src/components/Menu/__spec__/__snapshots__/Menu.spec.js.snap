// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders with default props 1`] = `
<ul
  className="root position_type_fixed"
/>
`;
